packages/react-server/src/server/test/server.test.tsx

```diff
@@ -16,37 +18,36 @@ jest.mock('@shopify/sewing-kit-koa', () => ({
 }));
 
 describe('createServer()', () => {
-  function MockApp() {
-    return <div>markup</div>;
-  }
-
-  let server: Server;
-  let port: number;
-  const ip = 'http://localhost';
-
-  beforeEach(async () => {
-    port = await getPort();
-
-    server = await createServer({
-      port,
-      ip,
-      render: () => <MockApp />,
-    });
-  });
-
-  afterEach(() => {
-    server.close();
+  afterAll(() => {
+    stopServers();
   });
 
   it('starts a server that responds with markup', async () => {
-    const response = await request(`${ip}:${port}`)
-      .get('/')
-      .then((resp: request.Response) => {
-        return resp;
-      });
+    function MockApp() {
+      return <div>markup</div>;
+    }
+
+    const response = await mountAppWithServer(<MockApp />);
 
     expect(response.text).toBe(
       `<!DOCTYPE html><html lang="en"><head><meta charSet="utf-8"/><meta http-equiv="X-UA-Compatible" content="IE=edge"/><meta name="referrer" content="never"/></head><body><div id="app"><div>markup</div></div></body></html>`,
     );
   });
+
+  it('supports updatable meta components', async () => {
+    const myTitle = 'Shopify Mock App';
```

Comment:
These tests are really nice *except* that they kind of obscure what we're actually testing. It might seem puritanical, but I think it's important to have the unit under test itself called directly in the test.

I've been thinking about how to do these node end-to-endy tests nicer and I think what we really want is something more like:

```
const rack = new TestRack();

beforeEach(() => {
  rack.unmountAll()
})

it('supports updatable meta components', async () => {
  const wrapper = rack.mount((port) => 
    createServer({port, render: () => <SomeJSX />})
  ) 

  const {text} = await wrapper.request('/').response()
  expect(response).toContainText('<title>whatever</title>')
```

where `TestRack` looks something like

```
interface TestRack {
  mount((port: number, ip: string) => Server): TestServer
  unmountAll(): void
}

interface ServerWrapper {
  request(fetchOptions?: FetchOptions):  Response
}
```

It's possible that `ServerWrapper` just wants to be a supertest(server), or a custom extended version of it, but I think the above is nice enough for now.